src/openai/_utils/_utils.py

```diff
@@ -412,3 +416,11 @@ def json_safe(data: object) -> object:
         return data.isoformat()
 
     return data
+
+
+def is_azure_client(client: object) -> TypeGuard[AzureOpenAI]:
+    return hasattr(client, "_azure_ad_token_provider")
```

Comment:
could this not be an `isinstance` check?

Reply:
I was struggling with circular imports, but I can just lazy import and make it an isinstance check